src/main/scala/com/codacy/client/bitbucket/v2/service/PullRequestServices.scala

```diff
@@ -17,14 +18,22 @@ class PullRequestServices(client: BitbucketClient) {
   def getPullRequests(
       owner: String,
       repository: String,
-      states: Seq[String] = Seq("OPEN"),
-      size: Int = 50
+      pageRequest: Option[PageRequest],
```

Comment:
Same as the other PR. Have you considered to have two methods instead of a method receiving `Option`?
Option is used here to distinguish between two different implementations instead of representing optional data.
If the paginated method is the one to use, I would create an override (maybe you can still have the one taking option as a private method for the implementation) with `pageRequest` and deprecate the old method with a nice message (like `Use the paginated override instead`)